lib/web_ui/lib/src/engine/mouse_cursor.dart

```diff
@@ -0,0 +1,57 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// @dart = 2.6
+part of engine;
+
+
+/// Provides mouse cursor bindings, such as the `flutter/mousecursor` channel.
+class MouseCursor {
+  /// Initializes the [MouseCursor] singleton.
+  ///
+  /// Use the [instance] getter to get the singleton after calling this method.
+  static void initialize() {
+    _instance ??= MouseCursor._();
+  }
+
+  /// The [MouseCursor] singleton.
+  static MouseCursor get instance => _instance;
+  static MouseCursor _instance;
+
+  MouseCursor._() {
+  }
+
+  static String _mapShapeCodeToCssValue(int shapeCode) {
+    // Shape codes are hard-coded identifiers for system cursors.
+    // 
+    // The shape code values must be kept in sync with flutter's
+    // rendering/mouse_cursor.dart
+    switch (shapeCode) {
+      case /* none */ 0x334c4a:
```

Comment:
Can you define kShapeCodeX/Y/Z in engine and then use those codes in rendering/mouse_cursor instead?

Reply:
Is it possible to import engine code from framework at all? Also these constants have to be replicated to other platforms as well, including Android, macOS, GLFW, etc.

Reply:
They would be part of dart:ui , just like StrokeJoin or other consts/enums. @gspencergoog : how do you typically handle consts across the 2 repos?

Reply:
Yeah, actually, I think Ferhat is right: we should define the codes just as contiguous integers, and define an enum that matches in the Dart code.

`StrokeJoin` defines an enum, and then in the C++/Java code, it defines constants that are used as indices into the `StrokeJoin.values` member.

Then the only things that need to be kept in sync are the enum (which is in the engine repo in dart:ui), and the constants (which are also in the engine repo).

You can also look at `ImageByteFormat` or `TileMode`, or any of the other enums we use that way in the engine.